pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -802,6 +802,12 @@ func applyLoadBalancer(cluster *apiv2.Cluster, lb *networking.LoadBalancerSettin
 		return
 	}
 
+	// Original destination service discovery must be used with the original destination load balancer.
+	if cluster.GetClusterDiscoveryType().Equal(&apiv2.Cluster_Type{Type: apiv2.Cluster_ORIGINAL_DST}) {
+		cluster.LbPolicy = apiv2.Cluster_ORIGINAL_DST_LB
+		return
```

Comment:
isn't this user error?

Reply:
we should make sure user fault can not result to system down

Reply:
@rshriram  The best way to prevent this is to do in validate stage, but this is caused by two different type of Config. So in order to make istio control plane more stable, have to do this tricky workaround.